src/libopenrave-core/jsonparser/jsonreader.cpp

```diff
@@ -362,6 +391,39 @@ class JSONReader
         }
     }
 
+    bool IsDownloadingFromRemote() const
+    {
+        return !_remoteUrl.empty();
+    }
+
+    std::string ResolveRemoteUri(const std::string &uri){
+        std::string scheme, path, fragment;
+        ParseURI(uri, scheme, path, fragment );
+
+
+        if (IsDownloadingFromRemote()) {
+
+            if (scheme == "file") {
+                return ("file://" + ResolveURI(scheme, path, std::string(), _vOpenRAVESchemeAliases));
+            }
+
+            return _remoteUrl + uri.substr(uri.find_last_of(":/")+1,  uri.find_last_of("#") - uri.find_last_of(":/") -1);
```

Comment:
Like I mentioned in the last round, `_remoteUrl` should only be used when you `curl_easy_setopt(CURLOPT_URL)`, it should not be used anywhere else in the code.